sdk/src/compute_budget_processor.rs

```diff
@@ -0,0 +1,153 @@
+//! Process compute_budget instructions outside of program-runtime to
+//! extract and sanitize limits.
+use crate::{
+    borsh0_10::try_from_slice_unchecked,
+    compute_budget::{self, ComputeBudgetInstruction},
+    entrypoint::HEAP_LENGTH as MIN_HEAP_FRAME_BYTES,
+    feature_set::{
+        add_set_tx_loaded_accounts_data_size_instruction, enable_request_heap_frame_ix,
+        remove_deprecated_request_unit_ix, FeatureSet,
+    },
+    genesis_config::ClusterType,
+    instruction::{CompiledInstruction, InstructionError},
+    pubkey::Pubkey,
+    sanitize::SanitizeError,
+    transaction_meta::{
+        TransactionMeta, DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT, MAX_COMPUTE_UNIT_LIMIT,
+        MAX_HEAP_FRAME_BYTES, MAX_LOADED_ACCOUNTS_DATA_SIZE_BYTES,
+    },
+};
+
+// Processing compute_budget could be part of tx sanitizing, failed to process
+// these instructions will drop the transaction eventually without execution,
+// may as well fail it early.
+// If succeeded, the transaction's specific limits/requests (could be default)
+// are retrieved and returned,
+pub fn process_compute_budget_instruction<'a>(
+    instructions: impl Iterator<Item = (&'a Pubkey, &'a CompiledInstruction)>,
+    feature_set: &FeatureSet,
+    maybe_cluster_type: Option<ClusterType>,
+) -> Result<TransactionMeta, SanitizeError> {
```

Comment:
it seems feasible that `TransactionMeta` will include more than just compute budget information in the future.

I wonder if we'd be better of having some sort of top-level fn for creating the `TransactionMeta`:

```rust
fn new_transaction_meta(...) -> Result<TransactionMeta, SanitizeError> {
    let mut meta = TransactionMeta::default();
    process_compute_budget_instrunction(..., &mut meta)?;
    check_is_simple_vote(..., &mut meta)?;
    ...
    meta
}
```

Reply:
^ on the fence about this exact structure. Potentially better to just have each internal fn return the parts needed, and then build the `Transactionmeta` at the end.

Reply:
totally. Ideally compute_budget_processor shouldn't need to know `TransactionMeta` at all. It could just return a tuple, or a struct?